packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,149 @@
+import type {
+	Message,
+	User,
+	TextChannel,
+	NewsChannel,
+	APIMessageContentResolvable,
+	MessageOptions,
+	MessageAdditions,
+	DMChannel,
+	CollectorFilter,
+	EmojiIdentifierResolvable,
+	MessageReaction
+} from 'discord.js';
+
+/**
+ * This is a [[MessagePrompter]], a utility to send a message with Yes/No promt.
+ * You must either use this class directly or extend it.
+ *
+ * [[MessagePrompter]] uses reactions to prompt for a yes/no answer and returns it.
+ * You can modify the confirm and cancel reaction used for each message, or use the [[MessagePrompter.defaultPrompts]].
+ * [[MessagePrompter.defaultPrompts]] is also static so you can modify these directly.
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you sure you want to continue?');
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you happy or sad?', {
+ * 		confirm: '🙂',
+ *		cancel: '🙁'
+ * });
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The message that will been sent in [[MessagePrompter.run]]
+	 */
+	public message: IMessagePrompterMessage | null = null;
+
+	/**
+	 * The message that has been sent in [[MessagePrompter.run]]
+	 */
+	public sentMessage: Message | null = null;
```

Comment:
Name it response

Reply:
I changed sentMessage to response, but later changed it to appliedMessage because response was more suited to the type: 'message' function. Which I added after your review. If you have a better name. Please tell me (: